### DISCUSSION TITLE
Get the status of Interactivity

### DISCUSSION THREAD

**Discussion by Djuppi** [xyflow/xyflow#4697] [Q&A] [answered]
Hi,

Im currently using xyflow with ReactJs for my company and i wondered if there is a way of getting the status of interactivity (if the lock control is locked or unlocked (true/false))? Would be nice to hide handles on my custom nodes when interactivity is false.

Thank you by the way for a fantastic library!!

**Comment by bcakmakoglu** [comment, marked as answer]
The lock state is defined by 3 values which you can select using `useStore` (same as the Controls do themselves tbh)

```ts
const selector = (s: ReactFlowState) => ({
  isInteractive: s.nodesDraggable || s.nodesConnectable || s.elementsSelectable,
});

// ...

const { isInteractive } = useStore(selector);
```

**Reply by Djuppi** [reply]
Thank you! Found out that I could use the isConnectable to hide/show handles. Hadn't thought of that for a solution.

**Comment by Djuppi** [comment]
Found out that I could use the isConnectable to hide/show handles. Hadn't thought of that for a solution.
